xray/plot/facetgrid.py

```diff
@@ -0,0 +1,383 @@
+from __future__ import division
+
+import warnings
+import itertools
+import functools
+
+import numpy as np
+
+from ..core.formatting import format_item
+from .plot import _determine_cmap_params
+
+
+# Overrides axes.labelsize, xtick.major.size, ytick.major.size
+# from mpl.rcParams
+_FONTSIZE = 'small'
+# For major ticks on x, y axes
+_NTICKS = 5
+
+
+def _nicetitle(coord, value, maxchar, template):
+    """
+    Put coord, value in template and truncate at maxchar
+    """
+    prettyvalue = format_item(value)
+    title = template.format(coord=coord, value=prettyvalue)
+
+    if len(title) > maxchar:
+        title = title[:(maxchar - 3)] + '...'
+
+    return title
+
+
+class FacetGrid(object):
+    """
+    Initialize the matplotlib figure and FacetGrid object.
+
+    The :class:`FacetGrid` is an object that links a xray DataArray to
+    a matplotlib figure with a particular structure.
+
+    In particular, :class:`FacetGrid` is used to draw plots with multiple
+    Axes where each Axes shows the same relationship conditioned on
+    different levels of some dimension. It's possible to condition on up to
+    two variables by assigning variables to the rows and columns of the
+    grid.
+
+    The general approach to plotting here is called "small multiples",
+    where the same kind of plot is repeated multiple times, and the
+    specific use of small multiples to display the same relationship
+    conditioned on one ore more other variables is often called a "trellis
+    plot".
+
+    The basic workflow is to initialize the :class:`FacetGrid` object with
+    the DataArray and the variable names that are used to structure the grid.
+    Then plotting functions can be applied to each subset by calling
+    :meth:`FacetGrid.map_dataarray` or :meth:`FacetGrid.map`.
+
+    Attributes
+    ----------
+    axes : numpy object array
+        Contains axes in corresponding position, as returned from
+        plt.subplots
+    fig : matplotlib.Figure
+        The figure containing all the axes
+    name_dicts : numpy object array
+        Contains dictionaries mapping coordinate names to values. None is
+        used as a sentinel value for axes which should remain empty, ie.
+        sometimes the bottom right grid
+
+    """
+
+    def __init__(self, darray, col=None, row=None, col_wrap=None,
+                 aspect=1, size=3):
+        """
+        Parameters
+        ----------
+        darray : DataArray
```

Comment:
Let's call this parameter `data`, like Seaborn. Eventually we might support dataset objects, too.